src/main/java/teammates/ui/controller/StudentCourseJoinAction.java

```diff
@@ -34,8 +35,13 @@ public ActionResult execute() throws EntityDoesNotExistException {
         
         String confirmUrl = Const.ActionURIs.STUDENT_COURSE_JOIN_AUTHENTICATED 
                 + "?" + Const.ParamsNames.REGKEY + "=" + regkey 
-                + "&" + Const.ParamsNames.NEXT_URL + "=" + nextUrl;
-        data = new StudentCourseJoinConfirmationPageData(account, student, confirmUrl, Logic.getLogoutUrl(confirmUrl));
+                + "&" + Const.ParamsNames.NEXT_URL + "=" + Sanitizer.sanitizeForNextUrl(nextUrl);
+        String nextUrlType = nextUrl.replaceFirst("^(/page/[A-Za-z]+)\\?.*$", "$1");
```

Comment:
It's good if this kind of thing can be unit tested. e.g. extract into a small static method and write a few unit tests against it.